src/Groups/types.jl

```diff
@@ -256,6 +258,31 @@ Group of automorphisms over a group of type `T`. It can be defined via the funct
   end
 end
 
+mutable struct AutomorphismGroupElem{T} <: GAPGroupElem{AutomorphismGroup{T}}
+  X::GapObj
+  parent::AutomorphismGroup{T}
+  map::fmpz_mat
+  AbstractAlgebra.@declare_other
+ 
+  function AutomorphismGroupElem{T}(F::GapObj, Aut::AutomorphismGroup{T}) where T
+    @assert F in Aut.X
+    z = new{T}(F,Aut,matrix(fmpz[]))
+    z.map =  hom(z).map
+    return z
+  end
+
+  
+end
+
+elem_type(Aut::AutomorphismGroup{T}) where T = AutomorphismGroupElem{T}
+
+group_element(Aut::AutomorphismGroup{T}, x::GapObj) where T = AutomorphismGroupElem{T}(x,Aut)
```

Comment:
I think turning this into
```
(Aut::AutomorphismGroup{T} where {T})(x::GapObj) = AutomorphismGroupElem{T}(x, Aut)
```
would be a good idea.

Reply:
This will automatically understand that  `elem_type` and  `group_element` are the one I have mentionned ? Here I did this to be able to use some of the already implemented method for GAPGroup. But, for instance, I had to re-implement the  `==` check since the structure is not working directly along with  `BasicGAPGroupElem`.

Reply:
I am just suggesting to replace `group_element(Aut, x)` by `Aut(x)`. Nothing else will change.

Reply:
Oh ok, I misunderstood. But even though, I have a small conflict then with  `Base.iterate` or  `Base.getindex` since both call  `group_elem` in their code. If I replace as you suggested, then the elements of my automorphism group are not of the type I have created.

Reply:
Can you show me which code is then failing? Just paste it here.

Reply:
Otherwise, paired with the global discussion, a solution would be to stick to the common `const AutomorphismGroupElem{T} where T = BasicGAPGroupElem{AutomorphismGroup{T}}` and decide not to store the matrix. For now it seems to be a solution, then we will have to see whether or not, in practice, it makes sense to store these matrices somewhere.